padl/transforms.py

```diff
@@ -931,6 +986,36 @@ def _pd_direct_subtransforms(self) -> Iterator[Transform]:
             if isinstance(v, Transform):
                 yield v
 
+    def _pd_get_non_target_stage_idx(self):
+        """Return an integer to track where a :class:`Compose` which failed got the Exception.
+
+        Example:
+            t = forward_1 >> unbatch >> post
+
+            Let's suppose we get an error on `post`, `post` is the index 1 of
+            `t.pd_postprocess`, then the element that fails on `t` is
+            t.pd_forward._pd_get_non_target_stage_idx() + 1 = 1 + 1 = 2
+        """
+        return 1
+
+    def _pd_get_target_stage_idx(self, is_entire_transform=None):
+        """Return an integer to track where a :class:`Compose` which failed got the Exception.
+
+        Example:
+            t = prep >> batch >> forward_1
+            Let's suppose we get an error on `forward_1`. `forward_1` is the index 0 of
+            `t.pd_forward`, then the element that fails on `t` is
+            2 + t.pd_forward._pd_get_stage_idx() = 2 + 0 = 2
+
+        :param is_entire_transform: *False* if *self* is not a part of a larger :class:`Transform`,
+            else *True*
+        """
+        return 0
+
+    def _pd_get_error_idx(self, stage: str):
```

Comment:
I thought it was clearer on `AtomicTransform` since different `Pipeline`s and `AtomicTransform` differ pretty much  and that method is only used on `AtomicTransform` and its subclasses, but I can change it.

Reply:
Right, I see. In that case you could also provide a `NotImplemented` version of the method in `Transform` class to make it clear to developers in the future that the method needs to be implemented for new internal Transforms that might not inherit from `AtomicTransform` or `Pipeline`

Reply:
ah sure, that sounds great